pkg/api/validate/dynamic/dynamic.go

```diff
@@ -197,27 +197,41 @@ func (dv *dynamic) validateVnetPermissions(ctx context.Context, vnet azure.Resou
 		"Microsoft.Network/virtualNetworks/subnets/write",
 	})
 
+	noPermissionsErr := api.NewCloudError(
+		http.StatusBadRequest,
+		errCode,
+		"",
+		"The %s service principal does not have Network Contributor permission on vnet '%s'.",
+		dv.authorizerType,
+		vnet.String(),
+	)
+
 	if err == wait.ErrWaitTimeout {
-		return api.NewCloudError(
-			http.StatusBadRequest,
-			errCode,
-			"",
-			"The %s service principal does not have Network Contributor permission on vnet '%s'.",
-			dv.authorizerType,
-			vnet.String(),
-		)
+		return noPermissionsErr
 	}
-	if detailedErr, ok := err.(autorest.DetailedError); ok &&
-		detailedErr.StatusCode == http.StatusNotFound {
-		return api.NewCloudError(
-			http.StatusBadRequest,
-			api.CloudErrorCodeInvalidLinkedVNet,
-			"",
-			"The vnet '%s' could not be found.",
-			vnet.String(),
-		)
+	if detailedErr, ok := err.(autorest.DetailedError); ok {
+		switch detailedErr.StatusCode {
+		case http.StatusNotFound:
+			return api.NewCloudError(
+				http.StatusBadRequest,
+				api.CloudErrorCodeInvalidLinkedVNet,
+				"",
+				"The vnet '%s' could not be found.",
+				vnet.String(),
+			)
+		case http.StatusForbidden:
+			return noPermissionsErr
```

Comment:
i think we should also include the detailed error that will provide more clarity

Reply:
This now includes the original error message.

Reply:
with checkAccessV2, will it return Forbidden at all?

Reply:
Yes, it can return forbidden as well but we're only going to use the FP to call it.  That's going to be a more of a fatal case.